### PR TITLE
Lint JS files with ESLint

### PR DESCRIPTION
**Pull request by mc10**
This is the beginning of some lint-based cleanup; we can enable more rules as we go on.
@azatoth: can you enable Travis builds on https://travis-ci.org/azatoth/twinkle?

### CODE REVIEW

**Comment by WMDE-Fisch** [comment]
:+1:  this codebase could need some more structure :-)

**Comment by azatoth** [comment]
@mc10 have activated travis now

File: modules/twinkleprod.js [hunk 1]
```diff
@@ -185,7 +182,7 @@ Twinkle.prod.callbacks = {
 
 		// Remove tags that become superfluous with this action
 		text = text.replace(/\{\{\s*(new unreviewed article|unreviewed|userspace draft|mtc|(copy|move) to wikimedia commons|(copy |move )?to ?commons)\s*(\|(?:\{\{[^{}]*\}\}|[^{}])*)?\}\}\s*/gi, "");
```
Comment (azatoth) [inline modules/twinkleprod.js]: should change above as well then I assume

File: modules/twinkleprotect.js [hunk 2]
```diff
@@ -1405,7 +1402,7 @@ Twinkle.protect.callbacks = {
 		var text = rppPage.getPageText();
 		var statusElement = rppPage.getStatusElement();
 
-		var rppRe = new RegExp( '===\\s*(\\[\\[)?\s*:?\s*' + RegExp.escape( Morebits.pageNameNorm, true ) + '\s*(\\]\\])?\\s*===', 'm' );
+		var rppRe = new RegExp( '===\\s*(\\[\\[)?\\s*:?\\s*' + RegExp.escape( Morebits.pageNameNorm, true ) + '\\s*(\\]\\])?\\s*===', 'm' );
```
Comment (azatoth) [inline modules/twinkleprotect.js]: one more \\s

Comment (mc10) [reply modules/twinkleprotect.js]: I think this one is okay; is there a specific `\s` you're referring to?

Comment (azatoth) [reply modules/twinkleprotect.js]: there's a \\s at the end of the regex as well

Comment (azatoth) [reply modules/twinkleprotect.js]: I read the diff backwards, I though you removed escapes :|

**Comment by MusikAnimal** [comment]
The removal of `<nowiki>` causes some code to get substituted when syncing to the wiki. Could you restore those? Surely you can do this and not upset the linter?

**Comment by MusikAnimal** [comment]
I do seem to recall someone saying <nowiki> is no longer needed for JS pages? Or maybe I'm making that up. Either way it definitely was substituting when I tried to sync, so I've created #413 and am syncing those versions. The code for #411 needs to go out today.

**Comment by mc10** [comment]
My bad, for some reason I thought it wasn't needed either. I think it's included for backwards compatibility purposes but can be disabled on a wiki if desired. The relevant bug [T34858](https://phabricator.wikimedia.org/T34858) seems like a won't-fix in the near future.

**Pull request merged**